### ISSUE TITLE
Local variables in functions are sometimes confused with code dependencies.

### ISSUE THREAD

**Issue by wlandau-lilly** [wlandau-lilly/drake#3]
```r
f = function(){
  y = 1
  return(y)
}

my_plan = plan(out = f())
make(my_plan) # correctly builds `out`
make(my_plan) # correctly skips `out`
y = 2 # gets confused as a dependency of f
make(my_plan) # incorrectly builds `out`. Not a disaster, but totally unnecessary.
readd(out) # correctly returns 1
```

I know I should be using codetools::findGlobals() rather than pryr::call_tree(). That way, local variables inside functions aren't confused with dependencies. I already solved this in the development branch, but the new code needs to be reviewed, and it will take some time to disclose it from my company.

**Comment by wlandau-lilly** [comment]
Fixed in version 2.0.0, which is unfortunately not public right now (see #5).

**Comment by wlandau-lilly** [comment]
Version 2.0.0 with the fix is disclosed.
